packages/protocol/contracts/L2/TaikoL2.sol

```diff
@@ -120,25 +182,53 @@ contract TaikoL2 is EssentialContract, TaikoL2Signer, IXchainSync {
 
         emit XchainSynced(l1Height, l1Hash, l1SignalRoot);
 
+        // Check EIP-1559 basefee
+        uint64 basefee;
+        if (gasIssuedPerSecond != 0) {
+            (basefee, gasExcess) = _calcBasefee(
+                block.timestamp - parentTimestamp,
+                uint64(block.gaslimit),
+                parentGasUsed
+            );
+        }
+
+        if (block.basefee != basefee)
+            revert L2_BASEFEE_MISMATCH(basefee, uint64(block.basefee));
+
+        parentTimestamp = uint64(block.timestamp);
+
         // We emit this event so circuits can grab its data to verify block variables.
         // If plonk lookup table already has all these data, we can still use this
         // event for debugging purpose.
+
         emit BlockVars({
-            number: block.number,
+            number: uint64(block.number),
+            basefee: basefee,
+            gaslimit: uint64(block.gaslimit),
+            timestamp: uint64(block.timestamp),
             parentHash: parentHash,
-            timestamp: block.timestamp,
-            basefee: 0, //block.basefee,
             prevrandao: block.prevrandao,
             coinbase: block.coinbase,
-            gaslimit: block.gaslimit,
-            chainid: block.chainid
+            chainid: uint32(block.chainid)
         });
     }
 
     /**********************
      * Public Functions   *
      **********************/
 
+    function getBasefee(
+        uint32 timeSinceNow,
```

Comment:
What is purpose of having the `timeSinceNow` ?
The name little bit confusing to me and also the fact that the `_getBasefeeAndPrint()` (which calls `getBasefee()` ) called everywhere with 0.

Reply:
You can get the basefee for a future time, assuming you will propose a block in 24 seconds, you can use 24 as the value of `timeSinceNow`.